cmd/snap/cmd_run.go

```diff
@@ -1225,6 +1289,11 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 			logger.Debugf("snap refreshes will not be postponed by this process")
 		}
 	}
+
+	if beforeExec != nil {
+		beforeExec()
+	}
```

Comment:
AFAICT, you are right, but I don't see how we can workaround this. @pedronis would love to know your opinion.

Reply:
This is the big question for me still, before approving.

Reply:
yes, it's about the cgroup, snapd unless forced will not inhibit the snap and do a refresh anymore because it detects app processes running for via the established cgroup, also notice that the lock would go away anyway on the exec as we open our locks with O_CLOEXEC